### PR TITLE
Expose ES7 compatible decorators and have full unit test coverage

### PR DESCRIPTION
**Pull request by tommyminds**

### CODE REVIEW

File: src/actions.js [hunk 1]
```diff
@@ -1,54 +1,120 @@
...
+      actionsName = `${clsName[0].toLowerCase()}${clsName.slice(1)}`;
     }
+
+    let namespace = isConfigObject && config.namespace !== undefined
+      ? config.namespace
+      : actionsName;
+
+    cls.actionNamespace = angular.isString(namespace) && namespace.length
+      ? namespace
+        .replace(/([\A-Z])/g, '_$1')
```
Comment (anton-107) [inline src/actions.js]: what is the reason for escaping A character here?

Comment (tommyminds) [inline src/actions.js]: Good catch!

File: src/annotation.js [hunk 2]
```diff
@@ -1,115 +1,113 @@
...
+  }
 
-    /**
-     * This method can be overriden by child classes to
-     * configure the angular module after it is created
-     * @param {module} module The created angular module
-     */
-    configure(module) {}
+  /**
+   * This method can be overriden by child classes to
```
Comment (anton-107) [inline src/annotation.js]: spelling: overridden

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import angular from 'angular';
+
+// Static object members need to be overriden async since some keys
```
Comment (anton-107) [inline src/utils.js]: spelling: overridden

File: src/annotation.js [hunk 4]
```diff
@@ -1,115 +1,113 @@
...
+  }
 
-    /**
-     * This method can be overriden by child classes to
-     * configure the angular module after it is created
-     * @param {module} module The created angular module
-     */
-    configure(module) {}
+  /**
+   * This method can be overridedn by child classes to
```
Comment (anton-107) [inline src/annotation.js]: spelling: overridden

File: src/store.js [hunk 5]
```diff
@@ -1,56 +1,104 @@
...
+    });
+  };
+}
+
+export function Handler(actionName) {
+  return (cls, handlerName) => {
+    let action = actionName;
+    if (!action) {
+      action = handlerName
+        .replace(/^on([A-Z]{1})/, (match, first) => first.toLowerCase())
```
Comment (anton-107) [inline src/store.js]: is it the same as /^on([A-Z])/ ? if yes, let's simplify

**Pull request merged**